docs/foundry/getting-started.md

```diff
@@ -0,0 +1,317 @@
+
+contract HelloChugSplash {
+    uint public number;
+    bool public stored;
+    address public otherStorage;
+    string public storageName;
+}
+```
+
+## 6. Create a ChugSplash file
+
+Next, we'll create a ChugSplash file, which contains all of the information necessary to deploy and upgrade your project. A ChugSplash file can be written in JavaScript or JSON. In this guide, it'll be a JSON file. ChugSplash files are the only files in your project that are not written in Solidity.
+
+In your project root:
+
+```
+mkdir chugsplash && echo > chugsplash/hello-chugsplash.json
+```
+
+Inside your newly created ChugSplash file, `hello-chugsplash.json`, copy and paste the following:
+
+```json
+{
+  "options": {
+    "projectName": "Hello ChugSplash"
+  },
+  "contracts": {
+    "HelloChugSplash": {
+      "contract": "HelloChugSplash",
+      "variables": {
+        "number": 1,
+        "stored": true,
+        "storageName": "First",
+        "otherStorage": "0x1111111111111111111111111111111111111111"
+      }
+    }
+  }
+}
+```
+
+We'll explain the details of the ChugSplash file in the next guide.
+
+## 7. Create your deployment script
+
+In the folder that contains your Foundry scripts (usually `script/`), create your deployment script. We'll call it `MyFirstProject.s.sol`.
+
+Inside, copy and paste the following:
+```sol
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+import "forge-std/Script.sol";
+import "chugsplash/ChugSplash.sol";
+
+// You *must* import the source files of all contracts you're deploying into your deployment script.
+import "../src/HelloChugSplash.sol";
+
+contract ChugSplashScript is Script {
+    function run() public {
+        // Create a ChugSplash instance
+        ChugSplash chugsplash = new ChugSplash();
+
+        // Define the path from the project root to your ChugSplash file.
+        string memory chugsplashFilePath = "./chugsplash/hello-chugsplash.json";
+
+        // Deploy all contracts in your ChugSplash file (in this case, just HelloChugSplash.sol)
+        chugsplash.deploy(chugsplashFilePath);
+    }
+}
+```
+
+Take a moment to read the comments in the file. In particular, note that it's required for you to import the source files of all the contracts you're deploying into your deployment script. For this project, the only file you need to import is `HelloChugSplash.sol`. This ensures that the latest artifacts of your contracts are included in the script. Don't worry if you forget to do this; we'll detect it and throw an error before the deployment is executed.
+
+## 8. Deploy with ChugSplash
+
+> Note: When deploying, upgrading, or testing your contracts locally, you must always use an Anvil node running as a stand-alone process.
+
+To create an Anvil node in a stand-alone process, run:
+
+```
+anvil
+```
+
+In another terminal window, run the following command to deploy your upgradeable contract:
+
+```
+forge script --rpc-url http://localhost:8545 script/MyFirstProject.s.sol
```

Comment:
We should either do `--rpc-url localhost` here or remove this from `foundry.toml`:
```
[rpc_endpoints]
localhost = "http://127.0.0.1:8545"
```

This is because the setting in foundry.toml is unused if we don't do `--rpc-url localhost`

Reply:
> the setting in foundry.toml is unused if we don't do --rpc-url localhost

Our library resolves the network url from the `foundry.toml` file using the network name fetched from the environment variable `NETWORK` or the default name `localhost`, so we need this value regardless. I suppose we could eliminate it by defaulting to `http://127.0.0.1:8545` if nothing is defined, but we'd still need to have the entry for all other networks so IMO we should maintain the pattern for localhost as well. 

I initially updated this because I don't think forge resolves `--rpc-url localhost` to the url defined in `foundry.toml`. I could be wrong about that though as it seems you disagree.

However, I just realized the `--rpc-url` parameter is actually not used at all because our library connects to the target network using the process described above. The only reason `--rpc-url` is needed is if the user wants to connect to the network from the forge environment in which case we recommend using `chugsplash.refresh()` to fork the target network which ensures that the transactions performed by our library outside the forge environment are replicated within it. 

So I think the correct answer here is actually to remove the `--rpc-url` param entirely.

Reply:
If we remove the `--rpc-url` flag and the user adds any transactions to the script that don't involve the `chugsplash` contract, the transactions won't go through. This is because foundry will use the in-process network for those transactions. That's why I originally added the `--rpc-url` flag

How about we just leave as is?